cli/src/semgrep/app/scans.py

```diff
@@ -366,35 +369,56 @@ def report_findings(
             logger.debug(f"Sending complete blob: {json.dumps(complete, indent=4)}")
 
         response = state.app_session.post(
-            f"{state.env.semgrep_url}/api/agent/scans/{self.scan_id}/findings_and_ignores",
+            f"{state.env.semgrep_url}/api/agent/scans/{self.scan_id}/results",
             timeout=state.env.upload_findings_timeout,
             json=findings_and_ignores,
         )
 
         try:
             response.raise_for_status()
 
-            resp_errors = response.json()["errors"]
+            res = response.json()
+            resp_errors = res["errors"]
             for error in resp_errors:
                 message = error["message"]
                 click.echo(f"Server returned following warning: {message}", err=True)
 
+            if "task_id" in res:
+                complete["task_id"] = res["task_id"]
+
         except requests.RequestException:
             raise Exception(f"API server returned this error: {response.text}")
 
-        # mark as complete
-        response = state.app_session.post(
-            f"{state.env.semgrep_url}/api/agent/scans/{self.scan_id}/complete",
-            timeout=state.env.upload_findings_timeout,
-            json=complete,
-        )
+        try_until = datetime.now() + timedelta(minutes=2)
```

Comment:
is 2 minutes enough?

Reply:
Good point - I bumped it to 10min which should be plenty, we can tune from there.